python/ray/ml/checkpoint.py

```diff
@@ -1,35 +1,94 @@
-import abc
-from typing import Dict, Optional
+import shutil
+import tarfile
+import tempfile
+
+import cloudpickle as pickle
+import os
+from typing import Optional
 
 import ray
+from ray.rllib.utils import DeveloperAPI
+from ray.util.ml_utils.cloud import (
+    upload_to_bucket,
+    is_cloud_target,
+    download_from_bucket,
+)
+
+
+DICT_CHECKPOINT_FILE_NAME = "dict_checkpoint.pkl"
+
+
+class Checkpoint:
```

Comment:
A naive question: Is this the unified API that could be used for RayTrain as well?

Reply:
Yes - this Checkpoint class will be used in all ML-related Ray libraries, i.e. Tune, Train, RLLib, and in the ML-related serving parts of Ray Serve

Reply:
I see.

Reply:
So we're going to remove `ray.train.Checkpoint` class and use `ray.ml.Checkpoint` instead?

Reply:
There is no `ray.train.Checkpoint` at the moment - only a `ray.train.PersistedCheckpoint`. We will consolidate the bookkeeping that is currently duplicated between Ray Train and Ray Tune (mostly in the `CheckpointManager` classes). The load/save functions of the respective trainables will continue to support their respective formats (Dicts and paths), for now we won't pass Checkpoints directly into them.